lib/MusicBrainz/Server/ControllerBase/WS/2.pm

```diff
@@ -310,6 +311,33 @@ sub make_list
     };
 }
 
+=head2 limit_releases_by_tracks
+
+Truncates a list of releases such that the entire list doesn't contain more
+than 500 tracks in total (but returns at least one release). The idea is to
+limit browse queries that contain an excessive number of tracks when
+C<inc=recordings> is specified.
+
+Note: This mutates the passed-in array reference C<$releases>.
+
+=cut
+
+sub limit_releases_by_tracks {
+    my ($self, $c, $releases) = @_;
+
+    my $track_count = 0;
+    my $release_count = 0;
+
+    for my $release (@{$releases}) {
+        $c->model('Medium')->load_for_releases($release);
+        $track_count += sum map { $_->track_count } $release->all_mediums;
+        last if $track_count > 500 && $release_count > 0;
```

Comment:
Just curious, how did you get to pick this particular number for the limit?
Would that be alright to define it as a constant or even as a DBDefs entry?